resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/PodTemplateConfigMapStep.scala

```diff
@@ -22,23 +22,39 @@ import java.nio.charset.StandardCharsets
 import com.google.common.io.Files
 import io.fabric8.kubernetes.api.model.{ConfigMapBuilder, ContainerBuilder, HasMetadata, PodBuilder}
 
-import org.apache.spark.deploy.k8s.{KubernetesConf, SparkPod}
+import org.apache.spark.deploy.k8s.{KubernetesConf, KubernetesUtils, SparkPod}
 import org.apache.spark.deploy.k8s.Config._
 import org.apache.spark.deploy.k8s.Constants._
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.{Clock, SystemClock}
 
-private[spark] class PodTemplateConfigMapStep(conf: KubernetesConf)
-  extends KubernetesFeatureConfigStep {
+private[spark] class PodTemplateConfigMapStep
+(conf: KubernetesConf, clock: Clock = new SystemClock())
+  extends KubernetesFeatureConfigStep with Logging {
 
   private val hasTemplate = conf.contains(KUBERNETES_EXECUTOR_PODTEMPLATE_FILE)
 
+  private val preferredConfigmapName = s"${conf.resourceNamePrefix}-$POD_TEMPLATE_CONFIGMAP"
+  private val resolvedConfigmapName =
+    if (preferredConfigmapName.length <= KUBERNETES_MAX_NAME_LENGTH) {
+    preferredConfigmapName
```

Comment:
indentation (line 39 ~ 40)?

Reply:
I think it is better now?

Reply:
By the way @stijndehaes I haven't been able to find the document that says the max name length of any configmap is 63 char long. From my quick command line test, I found it can be well over beyond that number. Could you confirm this limit is needed. Well, I do agree it is always good to be defensive but I found there is no such shortening logic for driver configmap either. So, could you confirm? Thanks.

Reply:
@yuj you are right the name is max 253 long according to this document: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/
I shall remove the logic, I don't think we need to be defensive when the max is this long.

Reply:
Well this makes the PR way shorter :)

Reply:
It's way better, isn't!